quinn/split_columns.py

```diff
@@ -0,0 +1,47 @@
+from typing import List
+from pyspark.sql import DataFrame
+from pyspark.sql.functions import col, split, when, length
+
+
+def split_col(df: DataFrame, col_name: str, delimiter: str,
+              new_col_names: List[str], mode: str = "permissive") -> DataFrame:
+    """
+    Splits the given column based on the delimiter and creates new columns with the split values.
+    :param df: The input DataFrame
+    :type df: pyspark.sql.DataFrame
+    :param col_name: The name of the column to split
+    :type col_name: str
+    :param delimiter: The delimiter to split the column on
+    :type delimiter: str
+    :param new_col_names: A list of two strings for the new column names
+    :type new_col_names: (List[str])
+    :param mode: The split mode. Can be "strict" or "permissive". Default is "strict"
```

Comment:
Looks like default now is permissive. But better to have strict here.

Reply:
Made changes to code as suggested.